src/common/components/formikPersist/FormikPersist.tsx

```diff
@@ -0,0 +1,58 @@
+import { FormikProps, useFormikContext } from 'formik';
+import debounce from 'lodash/debounce';
+import * as React from 'react';
+
+export interface PersistProps {
+  name: string;
+  debounceTime?: number;
+  isSessionStorage?: boolean;
+}
+
+const FormikPersist = ({
+  debounceTime = 300,
+  isSessionStorage = false,
+  name,
+}: PersistProps) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const formik = useFormikContext<any>();
+
+  const saveForm = React.useCallback(
+    debounce((data: FormikProps<{}>) => {
+      if (isSessionStorage) {
+        window.sessionStorage.setItem(name, JSON.stringify(data));
+      } else {
+        window.localStorage.setItem(name, JSON.stringify(data));
+      }
+    }, debounceTime),
+    [debounceTime, isSessionStorage, name]
+  );
+
+  React.useEffect(() => {
+    saveForm(formik);
+  }, [formik, saveForm]);
+
+  React.useEffect(() => {
+    let timeout: number;
+
+    const maybeState = isSessionStorage
+      ? window.sessionStorage.getItem(name)
+      : window.localStorage.getItem(name);
+
+    if (maybeState && maybeState !== null) {
```

Comment:
why is additional `maybeState !== null` check needed?

Reply:
No reason. Removed